### PR TITLE
Feature - Add nodeSelector, affinity and tolerations to daemonset and deployment templates

### PR DESCRIPTION
**Pull request by Dean-Coakley**
Fixes: https://github.com/nginxinc/kubernetes-ingress/issues/444

### Proposed changes
* Add nodeSelector support to deployment template
* Add affinity support to daemonset and deployment templates
* Tolerations changed to be more flexible. Can now be used in deployments and daemonsets that are not deployed to ICP
* Improved formatting of `values-icp.yaml`
### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/README.md [hunk 1]
```diff
@@ -71,7 +71,8 @@ Parameter | Description | Default
 `controller.defaultTLS.secret` | The secret with a TLS certificate and key for the default HTTPS server. The value must follow the following format: `<namespace>/<name>`. Used as an alternative to specifiying a certifcate and key using `controller.defaultTLS.cert` and `controller.defaultTLS.key` parameters. | None
 `controller.nodeSelector` | The node selector for pod assignment for the Ingress controller pods. | { }
 `controller.terminationGracePeriodSeconds` | The termination grace period of the Ingress controller pod. | 30
-`controller.tolerations` | The tolerations required for the IBM Cloud Private installation. | None
+`controller.tolerations` | The taints to tolerate for the Ingress controller pods. | []
```
Comment (pleshakov) [inline deployments/helm-chart/README.md]:
```suggestion
`controller.tolerations` | The tolerations of the Ingress controller pods. | []
```

File: deployments/helm-chart/README.md [hunk 2]
```diff
@@ -71,7 +71,8 @@ Parameter | Description | Default
...
+`controller.affinity` | The affinities of the ingress controller pods. | { }
```
Comment (pleshakov) [inline deployments/helm-chart/README.md]:
```suggestion
`controller.affinity` | The affinity of the Ingress controller pods. | { }
```

File: deployments/helm-chart/templates/controller-daemonset.yaml [hunk 3]
```diff
@@ -26,15 +26,15 @@ spec:
       terminationGracePeriodSeconds: {{ .Values.controller.terminationGracePeriodSeconds }}
 {{- if .Values.controller.nodeSelector }}
       nodeSelector:
-{{ toYaml .Values.controller.nodeSelector | indent 8 }}
+{{ toYaml .Values.controller.nodeSelector | trimSuffix "\n" | indent 8 }}
```
Comment (pleshakov) [inline deployments/helm-chart/templates/controller-daemonset.yaml]: here and in the other templates, I don't see a need to use `trimSuffix`.  that unnecessary newline doesn't hurt.

Comment (Dean-Coakley) [reply deployments/helm-chart/templates/controller-daemonset.yaml]: agreed. let's stay consistent with the rest of the template

**Pull request merged**